### DISCUSSION TITLE
Edge Fns Upload Image

### DISCUSSION THREAD

**Discussion by sdmotameni** [supabase/supabase#10468] [Questions] [answered]
Attempting to parse a request with multipart/form-data that contains an image and upload it to a storage bucket. I know I cannot write to fs with supabase edge fns but is their anyway other way I can achieve this? I attempted to create middleware, but am still experiencing issues.

**Comment by laktek** [comment, marked as answer]
We've put together an example showing how to upload files and add data into the database using a `multipart/form-data` request to a function https://github.com/supabase/supabase/blob/master/examples/edge-functions/supabase/functions/file-upload-storage/index.ts
